src/main/java/com/extremelyd1/game/chat/ChatChannelController.java

```diff
@@ -0,0 +1,30 @@
+package com.extremelyd1.game.chat;
+
+import org.bukkit.entity.Player;
+
+import java.util.HashMap;
+import java.util.Map;
+
+public class ChatChannelController {
+
+    public enum ChatChannel {
+        TEAM(), GLOBAL()
+    }
+
+    private static final ChatChannel DEFAULT_CHANNEL = ChatChannel.TEAM;
+
+    public Map<String, ChatChannel> playerChatChannels = new HashMap<>();
+
+    public void setPlayerChatChannel(Player player, ChatChannel channel) {
+        playerChatChannels.put(player.getUniqueId().toString(), channel);
+    }
+
+    public ChatChannel getPlayerChatChannel(Player player) {
+        ChatChannel channel = playerChatChannels.get(player.getUniqueId().toString());
+        if (channel == null) {

```

Comment:
Another remark about the style: you can use the `Map#containsKey` method instead of checking for `null`. In some cases          `null` values are allowed in a map and then this would be an issue, because you can't differentiate between whether there exists no mapping or it explicitly maps to `null`.

Reply:
of course, again this is my familiarity with null safety in Kotlin making me forget Java ways! I have modified this method.